### ISSUE TITLE
xBRZ support

### ISSUE THREAD

**Issue by Askmewho** [pjasicek/OpenClaw#163]
hi! I would love to try this game on xBRZ, would be a massive plus for this open source implementation. thanks!

**Comment by pjasicek** [comment]
Hi, what do you mean by xBRZ ? Like just using upscaled textures generated by xBRZ filter - https://sourceforge.net/projects/xbrz/ ?

**Comment by Askmewho** [comment]
on the fly, yes.

**Comment by AAide** [comment]
Hello, are you considering adding XBRZ? I already use it with the kega emulator and the quality with xbrz is really perfect its changes completely, it would be good if you seriously consider adding it.

**Comment by Askmewho** [comment]
I already use bilinear filtering from wine and it improves the visuals from the original game. But xbrz would be the right path!

**Comment by pjasicek** [comment]
No such plans since it's been some time I worked on this project. Anyway, it would not be as easy as just swapping the textures and it magically working. There are several issues:

- The level format revolves around having pixels as its coordinates - so if the textures would get upscaled, it would not suddenly worked
- My XML actor format uses pixels as dimensions
- The image format within Claw (*.PID) has x and y offsets for 2D animations - would not be **easily** possible with PNGs

**Comment by AAide** [comment]
> on the fly, yes.

Yes but its makes the game slightly blurry the bilinear, I could use the xbrz from reshad, but it's not really terrible, compare to xbrz on emulator, the quality is less good, even if I put xbrz4 which is the maximum that I have found for reshad

**Comment by pjasicek** [comment]
All I can do, directly with SDL2, is to use linear filtering:

Original (without linear filtering):

![image](https://user-images.githubusercontent.com/12572760/114244171-495a6a00-998e-11eb-9533-e9ceb9429e9b.png)


With linear filtering:

![image](https://user-images.githubusercontent.com/12572760/114244055-0dbfa000-998e-11eb-8445-1f1e7c242124.png)

**Comment by AAide** [comment]
Here is the best I got with reshade with the hqX4 shader 

without hqx4
![OpenClaw_b4Is9TujDv](https://user-images.githubusercontent.com/77096798/114245815-7c522d00-9991-11eb-97f8-2658b3c903f2.png)

with hqx4
![OpenClaw_3F0Ep1ORH4](https://user-images.githubusercontent.com/77096798/114245831-7fe5b400-9991-11eb-8aed-8c83cc509f6d.jpg)

my setting hqx4
![OpenClaw_sudUJhVktg](https://user-images.githubusercontent.com/77096798/114245839-8411d180-9991-11eb-9a03-fda78c88e0a7.png)

**Comment by Askmewho** [comment]
Can we have some samples with xbrz?? Yes, I noted you are not that active on the project anymore sir. Its okay. I would like to see this project finished with videos and xbrz/similar filter options. But its okay. I will find the way to improve a bit the original game. Thanks

**Comment by Askmewho** [comment]
Maybe scaleFX ? https://youtu.be/PdJXYA8Wmj4
